### PR TITLE
Fix flakey test for log file rotate.

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

File: daemon/logger/loggerutils/logfile_test.go [hunk 1]
```diff
@@ -266,13 +267,22 @@ func TestCheckCapacityAndRotate(t *testing.T) {
...
 	assert.NilError(t, l.WriteLogEntry(&logger.Message{Line: []byte("hello world!")}))
+	waitRotate(t, r, 30*time.Second)
 	_, err = os.Stat(f.Name() + ".1")
 	assert.NilError(t, err, dStringer)
+
+	// Compression is async, and we get a rotation event even though compression
+	// is not completed.
+	// However, the rotate lock is held until compression is completed.
+	// Lock the rotate mutex and check if the compressed file is there.
+	l.rotateMu.Lock()
```
Comment (thaJeztah) [inline daemon/logger/loggerutils/logfile_test.go:283]: "sgtm", but wondering if we're making the test too much tied to a side-effect of the implementation; would a "regular" retry loop be cleaner? Something like;

```go
func fileExists(name string) poll.Check {
	return func(t poll.LogT) poll.Result {
		_, err := os.Stat(name)
		switch {
		case err == nil:
			return poll.Success()
		case os.IsNotExist(err):
			return poll.Continue("waiting for %s to exist", name)
		default:
			return poll.Error(err)
		}
	}
}
```

And then

```go
poll.WaitOn(t, fileExists(f.Name()+".2.gz"), poll.WithTimeout(60*time.Second), poll.WithDelay(10*time.Second))
```

(same could be used for all the file checks in this test)

**Review by thaJeztah — COMMENTED** [review]
Left a suggestion / thought

File: daemon/logger/loggerutils/logfile_test.go (hunk 1 above)
Comment (cpuguy83) [reply daemon/logger/loggerutils/logfile_test.go:283]: Good point.

Comment (cpuguy83) [reply daemon/logger/loggerutils/logfile_test.go:283]: Fixed

**Review by thaJeztah — APPROVED** [review]
LGTM (if green) thanks for updating!

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**